### PR TITLE
Use intellij forms

### PR DESCRIPTION
**Pull request by uoftm**
This makes writing view code a lot easier, although dynamic view code is a bit trickier for some reason... You need to create a panel and then can append new things to the content of that panel

### CODE REVIEW

File: src/view/LoginView.java [hunk 1]
```diff
@@ -13,55 +12,31 @@
...
-
     this.loginController = controller;
     this.loginViewModel = loginViewModel;
     this.loginViewModel.addPropertyChangeListener(this);
 
-    JPanel body = new JPanel();
-    body.setLayout(new BoxLayout(body, BoxLayout.Y_AXIS));
-    // It seems the y-axis is ignored here
-    body.setMaximumSize(ViewConstants.paneSize);
+    contentPane.setBackground(ViewConstants.background);
```
Comment (alexrosen45) [inline src/view/LoginView.java:36]: Good

File: src/view/SignupView.form [hunk 2]
```diff
@@ -0,0 +1,185 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (alexrosen45) [inline src/view/SignupView.form:1]: We should consider creating a general form layout that we can pass fields, titles, etc. to, but this is fine for now.

File: src/view/SignupView.java [hunk 3]
```diff
@@ -112,16 +73,17 @@ public void keyPressed(KeyEvent e) {}
           @Override
           public void keyReleased(KeyEvent e) {}
         });
+
     // This makes a new KeyListener implementing class, instantiates it, and
     // makes it listen to keystrokes in the usernameInputField.
     //
     // Notice how it has access to instance variables in the enclosing class!
-    usernameInputField.addKeyListener(
```
Comment (alexrosen45) [inline src/view/SignupView.java:119]: Why rename these?

**Review by alexrosen45 — APPROVED** [review]

File: src/view/SignupView.java (hunk 3 above)
Comment (uoftm) [reply src/view/SignupView.java:119]: I was trying to make them more consistent with the button naming scheme, and it seemed clear enough that username was an input, but I'm happy to revert that change

File: src/view/SignupView.form (hunk 2 above)
Comment (uoftm) [reply src/view/SignupView.form:1]: Yeah, one of the tricky parts about intellij forms is definitely components... I tried to make chatview a component and failed pretty badly (nothing would render).

File: src/view/SignupView.java (hunk 3 above)
Comment (alexrosen45) [reply src/view/SignupView.java:119]: No, it's fine

File: src/view/SignupView.form (hunk 2 above)
Comment (alexrosen45) [reply src/view/SignupView.form:1]: hmm ok

**Pull request merged**